app/src/main/java/ch/epfl/sdp/WeatherFragment.java

```diff
@@ -0,0 +1,120 @@
+package ch.epfl.sdp;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
+
+import com.androdocs.httprequest.HttpRequest;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import ch.epfl.sdp.entity.PlayerManager;
+import ch.epfl.sdp.geometry.GeoPoint;
+
+public class WeatherFragment extends Fragment {
+    private TextView addressTxt, updated_atTxt, statusTxt, tempTxt, temp_minTxt, temp_maxTxt, sunriseTxt,
+            sunsetTxt, windTxt, humidityTxt;
+
+    private long sunrise, sunset;
+    private String updatedAtText, temp, tempMin,  tempMax, humidity, windSpeed, weatherDescription, address;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_weather, container, false);
+    }
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        addressTxt = getView().findViewById(R.id.address);
+        updated_atTxt = getView().findViewById(R.id.updated_at);
+        statusTxt = getView().findViewById(R.id.status);
+        tempTxt = getView().findViewById(R.id.temp);
+        temp_minTxt = getView().findViewById(R.id.temp_min);
+        temp_maxTxt = getView().findViewById(R.id.temp_max);
+        sunriseTxt = getView().findViewById(R.id.sunrise);
+        sunsetTxt = getView().findViewById(R.id.sunset);
+        windTxt = getView().findViewById(R.id.wind);
+        humidityTxt = getView().findViewById(R.id.humidity);
+
+        new WeatherTask().execute();
+    }
+
+    class WeatherTask extends AsyncTask<String, Void, String> {
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+
+            /* Showing the ProgressBar, Making the main design GONE */
+            getView().findViewById(R.id.loader).setVisibility(View.VISIBLE);
+            getView().findViewById(R.id.mainContainer).setVisibility(View.GONE);
+            getView().findViewById(R.id.errorText).setVisibility(View.GONE);
+        }
+
+        protected String doInBackground(String... args) {
+            GeoPoint location = PlayerManager.getCurrentUser().getLocation();
+            String API = "b840e4c524a759fd39f896b665f92de0";

```

Comment:
Maybe we should hide the API key, but I don't know if this important for this project.

Reply:
I thought about it, but I was not sure of how to do it, so I just let it here for now.